src/client/Microsoft.Identity.Client/Instance/Region/RegionManager.cs

```diff
@@ -81,9 +78,12 @@ public async Task<string> GetAzureRegionAsync(RequestContext requestContext)
                 requestContext.ApiEvent != null,
                 "Do not call GetAzureRegionAsync outside of a request. This can happen if you perform instance discovery outside a request, for example as part of validating input params.");
 
+            IRetryPolicyFactory retryPolicyFactory = requestContext.ServiceBundle.Config.RetryPolicyFactory;
+            IRetryPolicy retryPolicy = retryPolicyFactory.GetRetryPolicy(RequestType.STS);
```

Comment:
@bgavrilMS should we use the IMDS retry policy here with less number of retries? Maybe in another iteration?

Reply:
Makes sense, but it should be a separate PR.